Source/Evolve/WarpXEvolveEM.cpp

```diff
@@ -303,10 +304,29 @@ WarpX::OneStep_nosub (Real cur_time)
     FillBoundaryE();
     FillBoundaryB();
 #else
+    if (do_pml && pml_has_particles){
```

Comment:
Could you create a small function (similar to `DampJPML` below) to hide this code away?

Reply:
This is more complex than it seems since `pml` is in fact an array of pmls and not a pml itself: I cannot directly apply a method from the PML class